xds/src/main/java/io/grpc/xds/XdsNameResolver.java

```diff
@@ -147,7 +147,16 @@ final class XdsNameResolver extends NameResolver {
       XdsClientPoolFactory xdsClientPoolFactory, ThreadSafeRandom random,
       FilterRegistry filterRegistry, @Nullable Map<String, ?> bootstrapOverride) {
     this.targetAuthority = targetAuthority;
-    serviceAuthority = GrpcUtil.checkAuthority(checkNotNull(name, "name"));
+
+    // The name might have multiple slashes so encode it before verifying.
+    // If the encoding fails, fallback to the non-encoded string.
+    String authority = GrpcUtil.AuthorityEscaper.encodeAuthority(checkNotNull(name, "name"));
+    
+    // Verify the authority using encoding, but use non-escaped version for
+    // serviceAuthority.
+    GrpcUtil.checkAuthority(authority);
+    serviceAuthority = name;
```

Comment:
Uhh... you threw away `authority`.

Reply:
Fixed, when I initially drafted this PR my thought was to use the escaped version just for the check.